cattle/compute/__init__.py

```diff
@@ -63,6 +47,33 @@ def instance_deactivate(self, req=None, instanceHostMap=None,
                                                         progress)
         )
 
+    def get_instance_host_from_map(self, instanceHostMap=None):
+        instance = instanceHostMap.instance
+        host = instanceHostMap.host
+
+        try:
+            host.clusterConnection = \
+                instanceHostMap.data.fields['clusterConnection']
+            log.debug('clusterConnection = %s', host.clusterConnection)
+            if host.clusterConnection.startswith('https'):
+                try:
+                    host.caCrt = \
+                        instanceHostMap.data.fields['caCrt']
+                    host.clientCrt = \
+                        instanceHostMap.data.fields['clientCrt']
+                    host.clientKey = \
+                        instanceHostMap.data.fields['clientKey']
+                except (KeyError, AttributeError) as e:
+                    log.exception(
```

Comment:
Is there a specific reason why you chose to log and raise here? Typically, should log and handle without raising or just raise.

If not particular reason, I would suggest just raising an exception with the message that you are currently logging.